code/game/machinery/doors/windowdoor.dm

```diff
@@ -145,46 +167,29 @@
 	operating = 0
 	return 1
 
-/obj/machinery/door/window/take_damage(damage)
-	health = max(0, health - damage)
-	if(health <= 0)
-		var/debris = list(
-			new /obj/item/weapon/shard(loc),
-			new /obj/item/weapon/shard(loc),
-			new /obj/item/stack/rods(loc, 2),
-			new /obj/item/stack/cable_coil(loc, 2)
-			)
+/obj/machinery/door/window/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	switch(damage_type)
+		if(BRUTE)
+			playsound(loc, 'sound/effects/glasshit.ogg', 90, 1)
+		if(BURN)
+			playsound(loc, 'sound/items/welder.ogg', 100, 1)
+
+
+/obj/machinery/door/window/deconstruct(disassembled = TRUE)
+	if(can_deconstruct && !disassembled)
 		for(var/obj/fragment in debris)
+			fragment.forceMove(get_turf(src))
 			transfer_fingerprints_to(fragment)
-		density = 0
-		qdel(src)
-		return
+			debris -= fragment
+	qdel(src)
 
-/obj/machinery/door/window/bullet_act(obj/item/projectile/Proj)
-	if(Proj.damage)
-		if((Proj.damage_type == BRUTE || Proj.damage_type == BURN))
-			take_damage(round(Proj.damage / 2))
-	..()
+/obj/machinery/door/window/narsie_act()
+	color = NARSIE_WINDOW_COLOUR
 
-//When an object is thrown at the window
-/obj/machinery/door/window/hitby(atom/movable/AM)
+/obj/machinery/door/window/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
+	if(exposed_temperature > T0C + (reinf ? 1600 : 800))
+		take_damage(round(exposed_volume / 200), BURN, 0, 0)

```

Comment:
Are there any "reinforced" or plasma windoors? Because this could cause problems where they're used for atmos burn chambers or places with plasma glass.

Reply:
You can make windoors reinf ("secure") when building them by adding plasteel to the assembly to increase the temp tolerance, we don't have any plasma glass ones though although /vg/ has them so I could add them in for atmos uses